### PR TITLE
[ONL-7786] EcTextFilter

### PR DESCRIPTION
**Pull request by wolandec**

### CODE REVIEW

File: src/components/ec-text-filter/ec-text-filter.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      @icon-click="emit(TextFilterEvent.CHANGE, '')"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+
+import { IconName } from '../ec-icon/icon-names';
+import EcInputField from '../ec-input-field/ec-input-field.vue';
```
Comment (stavros-tomas) [inline src/components/ec-text-filter/ec-text-filter.vue:21]: Maybe `import EcInputField from '../ec-input-field';` ?

File: src/components/ec-text-filter/ec-text-filter.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    class="ec-text-filter"
+    :data-test="$attrs['data-test'] ? `${$attrs['data-test']} ec-text-filter` : 'ec-text-filter'"
+  >
+    <ec-input-field
+      v-model="inputModel"
+      v-bind="inputProps"
+      :left-icon="IconName.SimpleSearch"
+      :type="InputFieldType.TEXT"
+      :icon="rightIcon"
+      @icon-click="emit(TextFilterEvent.CHANGE, '')"
```
Comment (stavros-tomas) [inline src/components/ec-text-filter/ec-text-filter.vue:12]: I am not sure if this is a change event. Usually we have an an update function and an `update:modelValue` event. i.e https://github.com/Ebury/chameleon/blob/master/src/components/ec-currency-filter/ec-currency-filter.vue#L62

Comment (wolandec) [reply src/components/ec-text-filter/ec-text-filter.vue:12]: I add update:modelValue as well

File: src/components/ec-text-filter/ec-text-filter.story.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      <div class="tw-my-64 tw-mx-auto tw-max-w-screen-sm">
+        <ec-text-filter
+            v-bind="args"
+            v-bind:model-value="modelValue"
+            @change="handleChange"
+        />
+      </div>`,
+  }),
+  args: {
+    modelValue: 'Some text',
```
[the remark this answers is not in the document]
Comment (wolandec) [reply src/components/ec-text-filter/ec-text-filter.story.ts:45]: I added it in smart table story and use a placeholder

**Review by stavros-tomas — APPROVED** [review]

**Review by mcibique — APPROVED** [review]

**Review by stavros-tomas — APPROVED** [review]

**Pull request merged**